File: module-b-android/modules/aura-native/ios/AuraNativeModule.swift
import ExpoModulesCore

public class AuraNativeModule: Module {
  public func definition() -> ModuleDefinition {
    Name("AuraNative")

    AsyncFunction("detectObjectsAsync") { (_: String) -> [[String: Any]] in
      return []
    }

    AsyncFunction("startForegroundServiceAsync") { (_: String, _: String) in
      return
    }

    AsyncFunction("stopForegroundServiceAsync") {
      return
    }
  }
}

File: module-b-android/modules/aura-native/src/AuraNativeModule.ts
import { NativeModule, requireNativeModule } from 'expo';

import { AuraNativeModuleEvents, NativeDetectedObject } from './AuraNative.types';

declare class AuraNativeModule extends NativeModule<AuraNativeModuleEvents> {
  detectObjectsAsync(uri: string): Promise<NativeDetectedObject[]>;
  startForegroundServiceAsync(title: string, description: string): Promise<void>;
  stopForegroundServiceAsync(): Promise<void>;
}

export default requireNativeModule<AuraNativeModule>('AuraNative');
